syntax = "proto2";

package Attribute;

message Request {
  required int32 detect = 1;
  required int32 width = 2;
  required int32 high = 3;
  required bytes image = 4;
  required string requestid = 5;
  required double dtthres = 6;
  required double dtnms = 7;
  required int32 attr = 8;
  required int32 action = 9;
  required int32 cartype = 10;
  required int32 carmake = 11;
  required int32 maxObjNum = 12;
  required int32 attrW = 13;
  required int32 attrH = 14;
  required int32 nmsmethod = 15;
  required double nmssigma = 16;
  optional int32 track = 17;
}

message Response {
	required string requestid = 1;
	required Status status = 2;
	
	enum Status {
		RES_OK = 0;
		RES_ERROR = 1;
		RES_PARAMERROR = 2;
  }
  
	message Point {
		required int32 x = 1;
		required int32 y = 2;
	}

	message Detection {
		
		required Point p1 = 1;
		required Point p2 = 2;
		required Point p3 = 3;
		required Point p4 = 4;
		repeated Attr attrs = 5;
		required string obj_name = 6;
		required double score = 7;
		required int32 feature_dim = 8;
		repeated float feature_val = 9;
		required int32 trackID = 10;
		
		message Attr {
			required int32 name = 1;
			required double score = 2;
		}
		
	}
	repeated Detection detec = 3;
}